cpp/src/command_classes/CentralScene.cpp

```diff
@@ -201,10 +234,52 @@ bool CentralScene::HandleMsg
 
 		if( Node* node = GetNodeUnsafe() )
 		{
-				char lbl[64];
-				for (int i = 1; i <= m_scenecount; i++) {
-					snprintf(lbl, 64, "Scene %d", i);
-					node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), _instance, i, lbl, "", true, false, 0, 0 );
+				for (int i = 1; i <= m_scenecount && i <= 31; i++) {
+					if ( version2orHigher )
+					{
+					    if ( identical )
+					    {
+					        // instance for scene number normaly isn't right, but in this case i think this is the best solution.
+					        // Because i higly doubt that someone will ever use the multichannel class together with the centralScene class.
+                            int keyAttributes = _data[3];
+                            if ( keyAttributes & CentralSceneMask_KeyPressed1time)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_SinglePress, "Scene", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_KeyPressed2times)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_DoublePress, "Scene double press", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_HeldDown)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_HeldDown, "Scene held down", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_KeyReleased)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_Released, "Scene released", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_KeyPressed3times)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_TriplePress, "Scene triple press", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_KeyPressed4times)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_Pressed4Times, "Scene pressed 4 times", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_KeyPressed5times)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_Pressed5Times, "Scene pressed 5 times", "", true, false, 0, 0 );
+                            }
+                        }
```

Comment:
Sorry.. again jetlag hitting me.

But as noted above, using the instance is abusing the "API" :) We shouldn't do that, as then applications would need to implement code that behaves differently for CentralScene ValueID's versus other CC's.

I think it might be easier to do something like:
1) a List ValueID with the different key attributes. Everytime we get a updated Scene, we update the selected value. (including HeldDown/KeyReleased)

2) a Int ValueID that is updated with the Scene Number

3) a Button ValueID for HeldDown/KeyReleased events only

and when doing the logic in HandleMsg, make sure the List is updated first, then the Scene Number, and only update the Button if its a HeldDown/KeyReleased event.

(so a application can check the List for the type of Event via the List when it gets a updated Scene Number)